tiny_cnn/layers/convolutional_layer.h

```diff
@@ -406,7 +310,31 @@ class convolutional_layer : public feedforward_layer<Activation> {
         init();
     }
 
-private:
+ private:
+    void init_backend(backend_t backend_type) {
+        using namespace std::placeholders;  // for _1, _2, _3...
+        switch (backend_type) {
+            case backend_t::tiny_cnn:
+                Base::backend_ = std::make_shared<core::tiny_backend>(&params_,
+                    std::bind(&convolutional_layer::copy_and_pad_input, this, _1, _2),
+                    std::bind(&convolutional_layer::copy_and_unpad_delta, this, _1, _2),
+                    std::bind(&Base::backward_activation, this, _1, _2, _3),
+                    &conv_layer_worker_storage_);
+                Base::backend_->set_layer(this);
```

Comment:
Can you not use lambas here instead of `bind`? `std::bind` is pretty much discouraged, I believe it's [this talk](https://youtu.be/zt7ThwVfap0?t=25m8s) that is a good reference.